### DISCUSSION TITLE
Convert it to rust

### DISCUSSION THREAD

**Discussion by Eveeifyeve** [Everduin94/better-commits#90] [Ideas]
I've noticed it takes 2seconds to open up with rust you can use something like clap or argh to make a cli that is just as better as 2seconds more like instant.

**Comment by Eveeifyeve** [comment]
I am willing to make it.

**Comment by Everduin94** [comment]
Is it 2s for you every-time? 
Are you running the project in a semi-large repository?
Are you on a relatively fast, or slow, machine?

I don't think that typescript necessarily has a speed issue here. If I had to guess, it's the time to run things like `git rev-parse` and `git status` in a large project. On my slowest machine, it's still instant, so it's hard for me to test.

AFAIK Rust doesn't significantly speed up git operations like that. For example, I use starship as my terminal prompt (written in rust). It lags horribly when I repeatedly press enter and it keeps trying to fetch various git info.

**Comment by Eveeifyeve** [comment]
Depends how you develop it it can sometimes be faster than typescript in most cases like for an example lighting css.

**Comment by Eveeifyeve** [comment]
It's 2 seconds everytime and I am running project based on small repos and it's a fast machine.
